src/BaGet.UI/src/SearchResults.tsx

```diff
@@ -156,48 +156,68 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
             />
           </div>
         </form>
-        {this.state.items.map(value => (
-          <div key={value.id} className="row search-result">
-            <div className="col-sm-1 hidden-xs hidden-sm">
-              <img
-                src={value.iconUrl || DefaultPackageIcon}
-                className="package-icon img-responsive"
-                onError={this.loadDefaultIcon}
-                alt="The package icon" />
-            </div>
-            <div className="col-sm-11">
-              <div>
-                <Link to={`/packages/${value.id}`} className="package-title">{value.id}</Link>
-                <span>by: {value.authors}</span>
+        
+        {(() => {
+          if (this.state.items.length > 0) {
```

Comment:
Hm I noticed there's a small flaw with this logic. When we change the search input, we set `items` to an empty array, thereby causing that "Oops, nothing here..." message to flash while we load the new items. I think we should introduce a new `loading` property on the state object and change this line to:

```suggestion
          if (loading || this.state.items.length > 0) {
```

When we start loading new items using [`_loadItems`](https://github.com/loic-sharma/BaGet/blob/2ed9457a0c80e4755a224911f2509f44e8e5a8bf/src/BaGet.UI/src/SearchResults.tsx#L225), we should set `loading` to `true`. Once we complete loading items, we should set `loading` to `false`.

Reply:
Yeap, totally agreed